tests/stages/arxiv/test_arxiv_source.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+# SPDX-FileCopyrightText: Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import string
+import types
+from unittest import mock
+
+import pytest
+
+import cudf
+
+from _utils.dataset_manager import DatasetManager
+from morpheus.config import Config
+from morpheus.stages.input.arxiv_source import ArxivSource
+
+
+@mock.patch("langchain.text_splitter.RecursiveCharacterTextSplitter")
+def test_constructor(mock_splitter: mock.MagicMock, config: Config):
+    cache_dir = "/does/not/exist"
+    stage = ArxivSource(config, query="unittest", cache_dir=cache_dir, chunk_size=77, chunk_overlap=33, max_pages=100)
+    assert stage._query == "unittest"
```

Comment:
I don't think we should test internal state like this; doing so breaks notions of encapsulation, and makes tests more brittle because its possible to break them even if the external API is consistent and correct.